statistics/histogram.go

```diff
@@ -997,20 +1060,28 @@ func (idx *Index) MemoryUsage() (sum int64) {
 
 var nullKeyBytes, _ = codec.EncodeKey(nil, nil, types.NewDatum(nil))
 
-func (idx *Index) equalRowCount(sc *stmtctx.StatementContext, b []byte, modifyCount int64) (float64, error) {
+func (idx *Index) equalRowCount(b []byte, modifyCount int64) float64 {
 	if len(idx.Info.Columns) == 1 {
 		if bytes.Equal(b, nullKeyBytes) {
-			return float64(idx.NullCount), nil
+			return float64(idx.NullCount)
 		}
 	}
 	val := types.NewBytesDatum(b)
 	if idx.NDV > 0 && idx.outOfRange(val) {
-		return outOfRangeEQSelectivity(idx.NDV, modifyCount, int64(idx.TotalRowCount())) * idx.TotalRowCount(), nil
+		return outOfRangeEQSelectivity(idx.NDV, modifyCount, int64(idx.TotalRowCount())) * idx.TotalRowCount()
 	}
-	if idx.CMSketch != nil {
-		return float64(idx.QueryBytes(b)), nil
+	if idx.CMSketch != nil && idx.StatsVer == Version1 {
+		return float64(idx.QueryBytes(b))
+	}
+	// If it's version2, query the top-n first.
+	if idx.StatsVer == Version2 {
+		count, found := idx.TopN.QueryTopN(b)
+		if found {
+			return float64(count)
+		}
+		return idx.Histogram.equalRowCountV2(val)
 	}
-	return idx.Histogram.equalRowCount(val), nil
+	return idx.Histogram.equalRowCount(val)
```

Comment:
Should we use `idx.Histogram.BetweenRowCount(val, val+1)` here?
If the `Histogram` contains multiple columns, prefix point queries also use this function.
For example, the index is `key(a, b, c)`,  `where a=1 and b=1` also uses this function to estimate.